### PR TITLE
:arrow_up::one::zero: pos_scan_ref *Port*

### PR DESCRIPTION
**Pull request by Ommo73**

### CODE REVIEW

**Comment by yelizariev** [comment]
You must squash commit into one via ``git rebase -i``. Don't make new PR please

**Review by KolushovAlexandr — APPROVED** [review]

File: pos_scan_ref/static/src/js/pos.js [hunk 1]
```diff
@@ -1,52 +1,141 @@
...
+        var stored_categories = this.product_by_category_id;
 
-})();
\ No newline at end of file
+        if(!products instanceof Array){
+            products = [products];
+        }
+        for(var i = 0, len = products.length; i < len; i++){
+            var product = products[i];
+            var search_string = this._product_search_string(product);
```
[the remark this answers is not in the document]
Comment (KolushovAlexandr) [reply pos_scan_ref/static/src/js/pos.js]: Hound bot pick the wrong line. Look at the line below. This error about that

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]

File: pos_scan_ref/static/src/js/pos.js [hunk 2]
```diff
@@ -1,52 +1,85 @@
...
-})();
\ No newline at end of file
+PosDb.include({
+    init: function(options){
+        this._super(options);
+        this.product_by_reference = {};
+    },
+    add_products: function(products){
+        this._super(products);
+        var stored_categories = this.product_by_category_id;
```
Comment (KolushovAlexandr) [inline pos_scan_ref/static/src/js/pos.js]: Check the necessity of this stuff

File: pos_scan_ref/doc/index.rst [hunk 3]
```diff
@@ -1,3 +1,8 @@
-=====================================
- FIX searching product by ref in POS
-=====================================
+=======================================
+ Scan & search a product by ref in POS
+=======================================
+
+Installation
+============
```
Comment (KolushovAlexandr) [inline pos_scan_ref/doc/index.rst:6]: Add Usage

File: pos_scan_ref/README.rst [hunk 4]
```diff
@@ -1,5 +1,49 @@
...
+.. image:: https://img.shields.io/badge/license-LGPL--3-blue.png
+   :target: https://www.gnu.org/licenses/lgpl
+   :alt: License: LGPL-3
 
-Description: https://apps.odoo.com/apps/modules/8.0/pos_scan_ref
+=======================================
+ Scan & search a product by ref in POS
+=======================================
+
+Scan & search a product by ref in POS
```
Comment (KolushovAlexandr) [inline pos_scan_ref/README.rst]: No need to duplicate the module name

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]
1) :green_heart: commits say they add 'utf-8' line but i see they only delete the empty line beneath 'utf-8' comment. You can remove those commits
2) Don't forget to rebase
3) travis error

File: pos_scan_ref/doc/index.rst (hunk 3 above)
Comment (Ommo73) [reply pos_scan_ref/doc/index.rst:6]: Usage added

**Review by KolushovAlexandr — APPROVED** [review]

**Review by Ramil-Mukhametzyanov — APPROVED** [review]

**Pull request merged**